W-15161420 APIs for Anypoint Monitoring (was tracing and entities) (#272)
mulesoft/docs-monitoring: modules/ROOT/pages/_partials/amql.adoc

    filterExpression -> conditionExpresion [AND conditionExpression] 
    conditionExpression -> field comparisonOperator [literal|literList|field| NULL]
----

// end::filtering[]

// tag::grouping[]

[source,sql]
----
    aggregationExpression -> aggregationFunction(field[,parameters])
                    AggregationFunction = avg|sum|percentile|histogram|max|min|count
    aggregatedConditionExpression -> aggregationFunction(field[,parameters]) comparisionOperator [literal|literalList]                  
    havingConditionExpression -> aggregatedConditionExpression [ AND aggregatedConditionExpression]

----
// end::grouping[]

// tag::reservedKeywords[]

AMQL has several reserved keywords, which are not case sensitive. 

* `AS`
* `ASC`
* `BETWEEN`
* `BY`
* `DESC` 
* `FROM`
* `GROUP`
* `HAVING`
* `IN`
* `LIKE`
* `LIMIT`
* `OFFSET`
* `ORDER`
* `SELECT` 
* `WHERE`
* `TIMESTAMP`
* `TIMESERIES`

// end::reservedKeywords[]

// tag::selectClause[]
The `SELECT` clause enables you to get a list of fields in a data source using describe endpoints. 

When you use `SELECT`, consider the following:

* Use describe endpoints for each service to get a list of fields present in your data source.

> * Using `SELECT` with a wildcard (`select *`) returns all public endpoints for a data source, but note that not all data sources support it.

Reviewer: ```suggestion
* Using `SELECT` with a wildcard (`select *`) returns all public endpoints for a data source. Not all data sources support this option.
```

* Wrap string literals in single quotes.
* Field names in `SELECT` clauses are case-sensitive. 

> * Field alias is supported. The syntax is `field AS aliasName`.

Reviewer: ```suggestion
* Using a field alias is supported. The syntax is `field AS aliasName`.
```

* Wrap field names with double quotes to handle field names with special characters. The syntax is `select "field1", "field2.a" as 'fieldA' from dataSource`.

> * Aggregated fields support alias. The syntax is `aggregationFunction(field, ..) AS alias`. If an alias is not provided, the results return `aggregationFunction(field, ..)`.

Reviewer: ```suggestion
* Aggregated fields support aliases. The syntax is `aggregationFunction(field, ..) AS alias`. If an alias is not provided, the results return `aggregationFunction(field, ..)`.
```

> * Aggregated fields support alias. The syntax is `aggregationFunction(field, ..) AS alias`. If an alias is not provided, the results return `aggregationFunction(field, ..)`.

Reviewer: ```suggestion
* Aggregated fields support aliases. The syntax is `aggregationFunction(field, ..) AS alias`. If you don't provide an alias, the results return `aggregationFunction(field, ..)`.
```

`WHERE` supports the following data types:

* String literal
* Integer
* Decimal

`WHERE` uses the following operators:

|====
|Filter Function |Filter Operator |Notes

|`Equals`
|`=`
|

|`NotEquals`
|`!=`
|

|`greater`
|`>`
|

|`greaterOrEquals`
|`>=`
|

|`less`
|`<`
|

|`lessOrEquals`
|`+<=+`
|

|`in`
|`IN`
| in (...)

|`between`
|`BETWEEN`
a|The timestamp field supports only `between`. For example: 

[source,sql]
----
WHERE timestamp between 1710198204000 AND 1710371004000 
----

|`match pattern`
|`LIKE`
|This operator is a simplified version of the `match` operator, although it lacks regexp support. It supports only `wildcard`, noted as `*`.

|====
// end::whereClause[]

// tag::groupingClauseUR[]

The `GROUPING` clause enables you to aggregate numeric values using the following functions:

|====
|Aggregation |Arguments

|`avg`
|`<fieldName>`

|`sum`
|`<fieldName>`

|`percentile`
|`<fieldName, percentage value in range (0;100)>`

|`max`
|`<fieldName>`

|`min`
|`<fieldName>`

|`count`
|`<fieldName>`  or `<dimensionName>`

|`distinct_count`
|`<fieldName>` or `<dimensionName>`

|`histogram`
|`<fieldName, splitPointsArray>`

|====

> Note that some fields in the data might not support certain aggregations.

Reviewer: ```suggestion
Some fields in the data might not support certain aggregations.
```

// end::groupingClauseUR[]

// tag::timeseriesClause[]

> The TIMESERIES clause takes time window size returns events aggregated in provided time-buckets.

Reviewer: I don't understand this sentence. Is this referring to events for time window size returns?

Author: ```suggestion
The `TIMESERIES` clause takes the time window size and returns events that are aggregated in time buckets you specified.
```

> // tag::amqlIntro[]

Reviewer 2: This file (modules/ROOT/pages/_partials/amql.adoc) is included in the new am-apis.adoc
